app/controllers/reset_pins_controller.rb

```diff
@@ -3,59 +3,72 @@
 # Controller for reseting a Barcode+PIN user's PIN
 class ResetPinsController < ApplicationController
   before_action :logout_user!
+  rescue_from ActiveRecord::RecordNotFound, with: :user_not_found
```

Comment:
I still don't think we want to leak information about the presence + validity of a given library id?